parsers/verilog.c

```diff
@@ -1163,6 +1165,34 @@ static int skipParameterAssignment (int c)
 	return c;
 }
 
+// Using LRM Annex A Formal syntax, The declaration of function and task can be combined
+// to describe as follows (ignore parameter_value_assignment in the class_scope):
+// -------------------------------------------------------------------------------------
+//                       |  class_scope  |               |  class_scope  |
+// [ function | task ] [ { class_name :: } return_type ] { class_name :: } function_name
```

Comment:
You are right. In fact, it is not just that. If there is something like `logic [LOW:HIGH]`, the token `LOW` and `HIGH` will also be detected, but it will be skipped too.

This `[lifetime]` does not affect the execution of the algorithm. Considering that you have commented the complete function body below, I have omitted it here for simplicity. If added, it will be a duplicate of your complete comment below. However, I will add a supplementary comment later.

> Please add on the comment and add static or automatic on some complex pattens in your input.sv.

I will reply you at the location of the test code.

Reply:
Thank you.

Your test code was very easy to understand because the variable names were carefully chosen. I learned a lot.